### ISSUE TITLE
Serve a path named "upload"

### ISSUE THREAD

**Issue by KSXGitHub** [svenstaro/miniserve#151]
## Steps to reproduce

**Step 1:** Create a folder named `upload`

```sh
mkdir upload
```

**Step 2:** Launch `miniserve` with upload option

```sh
miniserve -u . -p 3333
```

**Step 3:** Open the browser, and go to http://127.0.0.1:3333/upload

## Expected behavior

It shows an empty directory.

## Actual behavior

> 404 Not Found
> Route /upload could not be found

![image](https://user-images.githubusercontent.com/11488886/60260749-74ccab80-9904-11e9-985c-acc00fc260cb.png)

## Suggestion

Don't use `/upload` route to upload files, instead, use `?function=upload` param. 

```html
<form action='?function=upload&path=/target' method='POST'>
```

---

**P.S.** I discovered this bug by inspecting upload form. There might be similar bugs somewhere.

**Comment by svenstaro** [comment]
Great find! Feeling up to fix this bug and provide a test case?

**Comment by KSXGitHub** [comment]
> Great find! Feeling up to fix this bug and provide a test case?

I am busy right now.

**Comment by aliemjay** [comment]
This issue is no longer valid. 
It has been fixed by updating to actix 2 (#343)

**Comment by svenstaro** [comment]
Indeed, just tested it. Thanks for reporting.
